src/main/java/core/CoreConstants.java

```diff
@@ -25,7 +25,7 @@
     public final static int backgroundImgHash = Hash.GetInstance().hash("backgroundImg");
 
     public enum GameEvents {
-        ABOUT_TO_START, GAME_OVER, ROUND_OVER, TURN_OVER, ACTION_CHOSEN, GAME_EVENT
+        ABOUT_TO_START, GAME_OVER, ROUND_OVER, TURN_OVER, ACTION_CHOSEN, ACTION_TAKEN, GAME_EVENT
```

Comment:
What's the difference between action chosen, and action taken?

Reply:
I added doc on this distintion to the point of their generation, but not here - I'll add it.

ACTION_CHOSEN = before action implemented (so state records information that affected the decision)
ACTION_TAKEN = after action implemented (which is what we want in order to update the GUI, say)